src/StatusTransitionsHandler.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+namespace RebelCode\Bookings\WordPress\Module;
+
+use Dhii\Exception\CreateInvalidArgumentExceptionCapableTrait;
+use Dhii\I18n\StringTranslatingTrait;
+use Dhii\Invocation\InvocableInterface;
+use Dhii\Util\Normalization\NormalizeArrayCapableTrait;
+use Psr\EventManager\EventInterface;
+use Traversable;
+
+/**
+ * Handler for providing the status related state and functionality.
+ *
+ * @since [*next-version*]
+ */
+class StatusTransitionsHandler implements InvocableInterface
+{
+    /* @since [*next-version*] */
+    use NormalizeArrayCapableTrait;
+
+    /* @since [*next-version*] */
+    use CreateInvalidArgumentExceptionCapableTrait;
+
+    /* @since [*next-version*] */
+    use StringTranslatingTrait;
+
+    /**
+     * Map of transitions that should be applied when changing status.
+     *
+     * @since [*next-version*]
+     *
+     * @var array
```

Comment:
Funny, the constructor accepts a `Traversable`, and sets this property to it without any checks or normalization; but the type of this property is `array`. How so?

Reply:
This is my mistake, I'll fix it.